File: Python/Kode_Program/Baju.py
from Aksesoris import Aksesoris

class Baju(Aksesoris):
    # Konstruktor untuk inisialisasi objek dengan atribut default
    def __init__(self, untuk="", size="", merk=""):
        self.Untuk = untuk  
        self.Size = size  
        self.Merk = merk  

    # Setter untuk mengubah nilai atribut 'Untuk'
    def Set_Untuk(self, untuk):
        self.Untuk = untuk  

    # Setter untuk mengubah nilai atribut 'Size'
    def Set_Size(self, size):
        self.Size = size  

    # Setter untuk mengubah nilai atribut 'Merk'
    def Set_Merk(self, merk):
        self.Merk = merk  

    # Getter untuk mengambil nilai atribut 'Untuk'
    def Get_Untuk(self):
        return self.Untuk  
    
    # Getter untuk mengambil nilai atribut 'Size'
    def Get_Size(self):
        return self.Size  
    
    # Getter untuk mengambil nilai atribut 'Merk'
    def Get_Merk(self):
        return self.Merk  
    
    # Menampilkan header tabel data produk baju
    @staticmethod
    def TampilHeader():
        print("+----+-----------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+")
        print("| ID | Nama Produk     | Harga       | Stok        | Jenis       | Ukuran      | Warna       | Untuk       | Size        | Merk        |")
        print("+----+-----------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+")

    # Menampilkan data baju dalam format tabel
    def TampilData(self):
        print(f"| {self.Get_ID():<2} | {self.Get_Nama()[:15]:<15} | Rp {self.Get_Harga():>8} | {self.Get_Stok():>11} | {self.Get_Jenis():<11} | {self.Get_Ukuran():<11} | {self.Get_Warna():<11} | {self.Untuk:<11} | {self.Size:<11} | {self.Merk:<11} |")
       
    # Menampilkan footer tabel data produk baju
    @staticmethod
    def TampilFooter():
        print("+----+-----------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+")

File: Java/Kode_Program/Baju.java
// Kelas Baju merupakan turunan dari kelas Aksesoris
// Kelas ini menyimpan informasi tambahan tentang baju hewan peliharaan
public class Baju extends Aksesoris {
    
    // Atribut tambahan khusus untuk baju
    private String untuk; // Jenis hewan yang cocok menggunakan baju ini (misalnya: Kucing, Kelinci, Burung)
    private String size;  // Ukuran baju (misalnya: S, M, L)
    private String merk;  // Merek baju

    // Konstruktor default (tanpa parameter), menginisialisasi atribut dengan nilai kosong
    public Baju() {
        this.untuk = "";
        this.size = "";
        this.merk = "";
    }

    // Konstruktor dengan parameter untuk langsung mengisi nilai atribut saat objek dibuat
    public Baju(String untuk, String size, String merk) {
        this.untuk = untuk;
        this.size = size;
        this.merk = merk;
    }

    // Getter method untuk mengambil nilai atribut "untuk"
    public String Get_Untuk() {
        return this.untuk;
    }

    // Getter method untuk mengambil ukuran baju
    public String Get_Size() {
        return this.size;
    }

    // Getter method untuk mengambil merk baju
    public String Get_Merk() {
        return this.merk;
    }

    // Setter method untuk mengubah nilai "untuk" (hewan yang cocok memakai baju)
    public void Set_Untuk(String Untuk) {
        this.untuk = Untuk;
    }

    // Setter method untuk mengubah ukuran baju
    public void Set_Size(String Size) {
        this.size = Size;
    }

    // Setter method untuk mengubah merk baju
    public void Set_Merk(String Merk) {
        this.merk = Merk;
    }

    // Method untuk menampilkan header tabel output
    public void TampilHeader() {
        System.out.println("+----+-----------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+");
        System.out.println("| ID | Nama Produk     | Harga       | Stok        | Jenis       | Bahan       | Warna       | Untuk       | Size        | Merk        |");
        System.out.println("+----+-----------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+");
    }
    
    // Method untuk menampilkan data produk dalam format tabel
    public void TampilData() {
        System.out.printf("| %2d | %-15s | %11d | %11d | %-11s | %-11s | %-11s | %-11s | %-11s | %-11s |\n", 
                          Get_ID(), Get_Nama(), Get_Harga(), Get_Stok(), Get_Jenis(), Get_Bahan(), Get_Warna(), Get_Untuk(), Get_Size(), Get_Merk());
    }

    // Method untuk menampilkan footer tabel output
    public void TampilFooter() {
        System.out.println("+----+-----------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+-------------+");
    }
}
